transport_plugins/jlink/iotile_transport_jlink/jlink_background.py

```diff
@@ -33,14 +40,22 @@ def __init__(self, jlink_adapter, loop=SharedLoop):
         self._maintenance_task = None
         self.rpc_response_event = self._loop.create_event() # triggered when the controller acknowledge rpc response
 
-    async def send_rpc(self, device_info, control_info, address, rpc_id, payload, timeout):
+    async def send_rpc(self, device_info, control_info, address, rpc_id, payload, timeout, best_effort=False):
         """Write and trigger an RPC."""
 
         write_address, write_data = control_info.format_rpc(address, rpc_id, payload)
         await self.write_memory(write_address, write_data, chunk_size=4)
 
         await self._trigger_rpc(device_info)
-        await self._wait_rpc_completion(control_info, timeout)
+
+        if rpc_id == self.RESET_RPC_ID and address == self.CONTROLLER_ADDRESS:
+            await asyncio.sleep(2)
+            await self._update_state_flags_after_reset(device_info, control_info)
+            return None
+        elif best_effort:
```

Comment:
It seems like this is a race condition: if an rpc with best effort is called followed by a normal rpc, the next rpc can be dispatched before the previous rpc finishes.  Given the intent, I believe, is to allow `notify_sensorgraph` to not hang `close_interface` if the device connection is lost, here's an alternative:

- remove `best_effort`
- wrap call to `notify_sensorgraph` in try/except to catch the timeout failure if the rpc fails.
- log that failure so that `notify_sensorgraph succeeds.

This removes the race and the need for special-casing inside `send_rpc`.

Reply:
You are right, I wasn't sure how to handle `close_interface` on reset, thank you for advise.